classes/news/NewsAPIPresenter.php

```diff
@@ -22,7 +22,32 @@ function (NewsPostDto $post) use ($system) {
                     'version' => $post->version,
                 ];
             },
-            $newsManager->getLatestPosts()
+            $newsManager->getNewsPosts($num_posts)
+        );
+    }
+
+    /**
+     * @param NewsPostDto[] $newsPosts
+     * @return array
+     */
+    public static function savePostResponse(NewsManager $newsManager, System $system, int $post_id, string $title, string $version, string $content, bool $update, bool $bugfix, bool $event, int $num_posts = 8): array {
+            $newsManager->saveNewsPost($post_id, $title, $version, $content, $update, $bugfix, $event);
```

Comment:
We should not have the presenter be changing any data, can you move this `saveNewsPost` call up to `api/news.php`?

Reply:
Fixed in commit.